### PR TITLE
Configuration refactor

### PR DESCRIPTION
**Pull request by pjagielski**
**Why this PR** By making the configuration nonstatic, there can be more than one review process engine running at a time. This paves the way for sputnik to work as a library rather than a jar executable. I admit that the way this is implemented here is a bit sloppy (configurations should all be in constructors, not methods!) but I feel it's a step in the right direction.

Note: this change will affect multiple method and constructor signatures. I recommend deploying a fresh version before this PR is merged.

### CODE REVIEW

**Comment by lbordowitz** [comment]
I ended up doing much the same changes on a plane ride, only to find that my PR had been closed and this PR, with a much better version of the same changes, popped up. 

+1, and I'll work on merge conflicts so that config moves from method signature to constructor signature.

File: src/main/java/pl/touk/sputnik/Main.java [hunk 1]
```diff
@@ -29,20 +29,19 @@ public static void main(String[] args) {
...
     }
 
     private static ConnectorFacade getConnectorFacade(Configuration configuration) {
         ConnectorType connectorType = ConnectorType.getValidConnectorType(configuration.getProperty(GeneralOption.CONNECTOR_TYPE));
         ConnectorFacade facade = null;
         try {
-            facade = ConnectorFacadeFactory.INSTANCE.build(connectorType);
-            facade.validate(ConfigurationHolder.instance());
+            facade = ConnectorFacadeFactory.INSTANCE.build(connectorType, configuration);
+            facade.validate(configuration);
```
Comment (lbordowitz) [inline src/main/java/pl/touk/sputnik/Main.java]: Do we need to validate, have a validate function? Maybe the builder would be the best place to put validation logic, and we can package-protect facade constructors to ensure that these class instances are created by builders. This is a bit of scope-creep, so it's an optional improvement.

**Comment by lbordowitz** [comment]
Gave the PR a once-over, and it looks good. I still have not tried it in action (i.e. running sputnik on a repository) and would feel much more comfortable about this being merged once I do so.

**Comment by pjagielski** [comment]
:+1:  from me for the code, tested it also on some basic project and looks fins

**Comment by SpOOnman** [comment]
@pjagielski merge it!

**Comment by pjagielski** [comment]
@lbordowitz thanks again!

**Comment by lbordowitz** [comment]
My pleasure! If we get a new maven release (1.5.0), I'll even be able to finish that Jenkins plugin I've been working on.

**Pull request merged**